src/streamlink/plugins/vk.py

```diff
@@ -11,10 +11,10 @@
 
 
 class VK(Plugin):
-    _url_re = re.compile(r"http(?:s)?://(\w+\.)?vk.com/video-[0-9]*_[0-9]*")
-    _url_catalog_re = re.compile(r"http(?:s)?://(\w+\.)?vk.com/videos-[0-9]*")
-    _livestream_sources_re = re.compile(r"<source src=\\\"(.*?)\\\" type=\\\"application\\\/vnd\.apple\.mpegurl\\\">")
-    _vod_sources_re = re.compile(r"<source src=\\\"(.*?)\\\" type=\\\"video\\\/mp4\\\">")
+    _url_re = re.compile(r"https?://(\w+\.)?vk.com/(video\?z\=)?video-?[0-9]*_[0-9]*")
+    _url_catalog_re = re.compile(r"https?://(\w+\.)?vk.com/videos-[0-9]*(?:_[0-9]*)?")
+    _livestream_sources_re = re.compile(r"src=(?:\\)?\"((?!https?:(?:\\)?/(?:\\)?/vk\.com(?:\\)?/video_hls\.php)[^\"']+?)\" type=(?:\\)?\"application(?:\\)?\/vnd\.apple\.mpegurl(?:\\)?\"")
+    _vod_sources_re = re.compile(r"<source src=(?:\\)?\"(https?:(?:\\)?/(?:\\)?/cs.*(?:\\)?)\" type=(?:\\)?\"video(?:\\)?\/mp4(?:\\)?\" (?:\\)?/>")
```

Comment:
you don't need to ignore `\` with `(?:\\)?`

it's always there

```
type=\"video\/mp4\" \/>
```

---

and the regex doesn't work correctly because of `.*` after cs
r
un your regex on this it will only find one long false url

```
<source src=\"https:\/\/cs63.vkuservideo.net\/1\/1\/videos\/1.720.mp4?extra=id\" type=\"video\/mp4\" \/><source src=\"https:\/\/cs63.vkuservideo.net\/1\/1\/videos\/1.480.mp4?extra=id\" type=\"video\/mp4\" \/><source src=\"https:\/\/cs632205.vkuservideo.net\/1\/1\/videos\/1.360.mp4?extra=id\" type=\"video\/mp4\" \/><source src=\"https:\/\/cs632205.vkuservideo.net\/1\/1\/videos\/1.240.mp4?extra=id\" type=\"video\/mp4\" \/>
```